tests/unit/controlgroup/core.js

```diff
@@ -160,4 +160,42 @@ QUnit.test( "Single controlgroup button - vertical", function( assert ) {
 		" ui-corner-tr ui-corner-tl ui-corner-bl ui corner-br" );
 } );
 
+QUnit.module( "Controlgroup: Non-empty class key", {
+	setup: function() {
+		this.classKey = $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ];
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] =
+			$.trim( $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] +
+				" something-custom" );
+	},
+	teardown: function() {
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] = this.classKey;
+	}
+} );
+
+QUnit.test( "Controlgroup instantiates child widgets with correct options", function( assert ) {
+	assert.expect( 1 );
+
+	$( ".controlgroup-non-empty-class-key-init" ).controlgroup();
+
+	assert.hasClasses( $( "#non-empty-class-key-init-child" ).next(), "something-custom" );
+} );
+
+QUnit.test( "Controlgroup correctly assigns child widget classes options key", function( assert ) {
+	assert.expect( 2 );
+
+	$( ".controlgroup-non-empty-class-key-value-no-dupe" ).controlgroup();
+
+	assert.strictEqual(
+		( $( "#non-empty-class-key-value-no-dupe-first" )
+			.selectmenu( "option", "classes" )[ "ui-selectmenu-button-closed" ]
+			.match( /something-custom/ ) || [] ).length, 1,
+		"Class 'something-custom' appears exactly once in the first widget's class key value" );
+
+	assert.strictEqual(
+		( $( "#non-empty-class-key-value-no-dupe-second" )
+			.selectmenu( "option", "classes" )[ "ui-selectmenu-button-closed" ]
+			.match( /something-custom/ ) || [] ).length, 1,
+		"Class 'something-custom' appears exactly once in the second widget's class key value" );
+} );
```

Comment:
can we shorten all the class and id's here and in the html kind of excessively long